src/main/java/net/neoforged/neoforge/capabilities/BlockCapability.java

```diff
@@ -120,11 +122,46 @@ public final class BlockCapability<T, C extends @Nullable Object> extends BaseCa
 
     /**
      * {@return a new immutable copy of all the currently known block capabilities}
+     *
+     * <p>Mods that want to forward "all" capability requests should likely use {@link #getAllProxyable()} instead.
      */
     public static synchronized List<BlockCapability<?, ?>> getAll() {
         return registry.getAll();
     }
 
+    /**
+     * {@return a new immutable copy of all the currently known proxyable block capabilities}
+     *
+     * <p>This method should typically only be used in the {@link EventPriority#LOW} or
+     * {@link EventPriority#LOWEST} phase of {@link RegisterCapabilitiesEvent}, or later,
+     * to ensure that all proxyable capabilities have been {@link RegisterCapabilitiesEvent#setProxyable marked as such}.
+     *
+     * @see #isProxyable()
+     */
+    public static synchronized List<BlockCapability<?, ?>> getAllProxyable() {
+        return registry.getAll().stream().filter(BlockCapability::isProxyable).toList();
+    }
+
+    /**
+     * Returns whether this capability is proxyable.
+     * This information is metadata: it does not change how the capability works internally,
+     * but it tells mods whether they should or should not register proxying capability providers.
+     *
+     * <p>If the capability is proxyable, requests for this capability are safe to forward unilaterally to other blocks.
+     *
+     * <p>If the capability is not proxyable, requests for this capability should not be forwarded to other blocks without further information.
+     * In that case, refer to documentation of the capability to understand under which circumstances it is safe to forward, if at all.
+     * For this reason, mods that forward "all" capabilities should not forward non-proxyable capabilities.
+     *
+     * <p>Block capabilities are not proxyable by default.
+     * Any call to {@link RegisterCapabilitiesEvent#setProxyable(BlockCapability)} will mark the capability as proxyable.
+     * Any call to {@link RegisterCapabilitiesEvent#setNonProxyable(BlockCapability)} will mark the capability as non-proxyable,
```

Comment:
I am a little unsure of how useful having this be a TriState is given we say that only the capability creator should mark it as proxyable. Is this to allow mods to defensively mark caps that crash as non-proxyable to avoid any mods that might allow users to configure specific capabilities as proxyable in some way or other?

Reply:
The goal is twofold:
- For non-proxyable caps, make sure that they cannot be marked as proxyable by another mod as that would cause crashes.
- For proxyable caps, make sure that they cannot be marked as non-proxyable by another mod as that would break gameplay in subtle ways (e.g. some random mod making `IItemHandler` non-proxyable).